mzLib/Transcriptomics/Classes/NucleicAcid.cs

```diff
@@ -0,0 +1,393 @@
+        private IHasChemicalFormula _3PrimeTerminus;
+
+        /// <summary>
+        /// All of the modifications indexed by position from 5- to 3-prime termini. This array is 2 bigger than the nucleic acid array
+        /// as index 0 and Count - 1 represent the 5- and 3-prime terminus, respectively
+        /// </summary>
+        private IHasMass[] _modifications;
+
+        /// <summary>
+        /// All of the nucleic acid residues indexed by position from 5- to 3-prime.
+        /// </summary>
+        private Nucleotide[] _nucleicAcids;
+
+        /// <summary>
+        /// The nucleic acid sequence with modification names interspersed. Is ignored if 'StoreSequenceString' is false
+        /// </summary>
+        private string _sequenceWithMods;
+
+        /// <summary>
+        /// The nucleic acid sequence. Is ignored if 'StoreSequenceString' is false
+        /// </summary>
+        private string _sequence;
+
+        #endregion
+
+        #region Internal Properties
+
+        /// <summary>
+        /// The internal data store for the modifications (2 larger than the length to handle the 5' and 3' termini)
+        /// </summary>
+        internal IHasMass[] Modifications { get; }
+
+        /// <summary>
+        /// The internal data store for the nucleic acids
+        /// </summary>
+        internal Nucleotide[] NucleicAcids => _nucleicAcids;
+
+        #endregion
+
+        #region Public Properties
+
+        /// <summary>
+        /// Gets or sets the 5' terminus of this nucleic acid polymer
+        /// </summary>
+        public IHasChemicalFormula FivePrimeTerminus
+        {
+            get => _5PrimeTerminus;
+            set => ReplaceTerminus(ref _5PrimeTerminus, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the 3' terminus of this nucleic acid polymer
+        /// </summary>
+        public IHasChemicalFormula ThreePrimeTerminus
+        {
+            get => _3PrimeTerminus;
+            set => ReplaceTerminus(ref _3PrimeTerminus, value);
+        }
+
+        /// <summary>
+        /// Gets the number of nucleic acids in this nucleic acid polymer
+        /// </summary>
+        public int Length { get; private set; }
+
+        /// <summary>
+        /// The total monoisotopic mass of this peptide and all of its modifications
+        /// </summary>
+        public double MonoisotopicMass { get; private set; }
+
+        /// <summary>
+        /// Returns a copy of the nucleic acid array, used for -base mass calculations.
+        /// </summary>
+        public Nucleotide[] NucleicAcidArray => _nucleicAcids;
+
+        #endregion
+
+        #region Nucleic Acid Sequence
+
+        /// <summary>
+        /// Gets the base nucleic acid sequence
+        /// </summary>
+        public string BaseSequence
+        {
+            get
+            {
+                // Generate the sequence if the stored version is null or empty
+                if (string.IsNullOrEmpty(_sequence))
```

Comment:
do we need a check to make sure that the sequence that gets passed is actually composed of acceptable characters? I ran into this problem elsewhere.